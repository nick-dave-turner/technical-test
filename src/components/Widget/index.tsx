import React from "react";
import styled from "styled-components";

const Container = styled.div``;

const Header = styled.div`
  padding: 0.625rem 1.25rem;
  background: #999999;
  border-radius: 0.625rem 0.625rem 0 0;
`;

const Body = styled.div`
  padding: 1.25rem;
  background: #ffffff;
  border-radius: 0 0 0.625rem 0.625rem;
`;

const Title = styled.h4`
  margin: 0;
  color: #ffffff;
`;

type WidgetProps = {
  children: React.ReactChild;
  title?: string;
};

const Widget: React.FC<WidgetProps> = ({ children, title }) => {
  return (
    <Container>
      {title && (
        <Header>
          <Title>{title}</Title>
        </Header>
      )}
      <Body title={title}>{children}</Body>
    </Container>
  );
};

export default Widget;
