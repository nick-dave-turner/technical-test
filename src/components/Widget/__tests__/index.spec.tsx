import React from "react";
import { shallow } from "enzyme";

import Widget from "../";

describe("Widget component", () => {
  describe("When Widget title present", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <Widget>
          <h1>Hello World</h1>
        </Widget>
      );
      expect(component).toMatchSnapshot();
    });
  });

  describe("When Widget title not present", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <Widget title="TESTING">
          <h1>Hello World</h1>
        </Widget>
      );
      expect(component).toMatchSnapshot();
    });
  });
});
