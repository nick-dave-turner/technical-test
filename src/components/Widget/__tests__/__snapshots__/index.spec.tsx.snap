// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget component When Widget title not present should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.h4>
      TESTING
    </styled.h4>
  </styled.div>
  <styled.div
    title="TESTING"
  >
    <h1>
      Hello World
    </h1>
  </styled.div>
</styled.div>
`;

exports[`Widget component When Widget title present should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <h1>
      Hello World
    </h1>
  </styled.div>
</styled.div>
`;
